### PR TITLE
document tracker exploration: initial scaffolding

### PR DESCRIPTION
**Pull request by fontanierh**
Functional no-op until the `SHOULD_RUN_POST_UPSERT_HOOKS` is set

### CODE REVIEW

**Comment by spolu** [comment]
Exciting!

File: connectors/src/post_upsert_hooks/index.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { documentTrackerPostUpsertHook } from "@connectors/post_upsert_hooks/document_tracker";
+import { DataSourceConfig } from "@connectors/types/data_source_config";
+
+export type PostUpsertHook = (
+  dataSourceConfig: DataSourceConfig,
+  documentId: string
+) => Promise<void>;
+
+export const POST_UPSERT_HOOKS: PostUpsertHook[] = [
```
[the remark this answers is not in the document]
Comment (spolu) [reply connectors/src/post_upsert_hooks/index.ts:9]: Concluding here :)

Let's just add an explicit comments on the expectactions for these functions

**Review by spolu — DISMISSED** [review]
LGTM

Left a few comments describing my thinking => conclusion let's just add a comment

File: connectors/src/post_upsert_hooks/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fontanierh) [reply connectors/src/post_upsert_hooks/index.ts:9]: Interesting, I like it.

Maybe 

```typescript
type PostUpsertHook = {filter: PostUpsertHookFilter, workflow: PostUpsertHookWorkflow}
```

?

[the remark this answers is not in the document]
Comment (spolu) [reply connectors/src/post_upsert_hooks/index.ts:9]: Ah yeah that will make it explicit that we launch a workflow but you keep control of the workflow it's just that the arguments are standardized

Bullish let's do this!

[the remark this answers is not in the document]
Comment (spolu) [reply connectors/src/post_upsert_hooks/index.ts:9]: :100:

File: connectors/src/post_upsert_hooks/index.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+// Should return true if the post upsert hook should run for this document
+// Should return false if the post upsert hook should not run for this document
+// Should be relatively quick to run
+// Will not be retried
+export type PostUpsertHookFilter = (
+  dataSourceConfig: DataSourceConfig,
+  documentId: string
+) => Promise<boolean>;
+
+export type PostUpsertHook = {
```
Comment (spolu) [inline connectors/src/post_upsert_hooks/index.ts:24]: Naice

**Pull request merged**